UI/DamageIndicator.cs

```diff
@@ -0,0 +1,42 @@
+using Godot;
+
+
+public class DamageIndicator : TextureRect
+{
+    Vector3 ShotFirePosition;
+	Vector2 FlatShotFirePos;
+	float PlayerStartRotation;
+
+
+	public override void _Ready()
+	{
+		// Start transparency Tween in _Ready so we know the Tween has been created
+		Tween TransparencyTween = GetNode<Tween>("Tween");
```

Comment:
Lets create the tween instead of getting it since we're already setting all of its "setting" in code

Reply:
I'm a dope, ignore this. I feel like an asshole saying this now but we don't want a tween. I'll explain later sorry

Reply:
We don't want a tween because of several reasons:
* Requires babysitting a non-typechecked (or runtime type checked) interface
* The finish callback relies on a string to know what function to call which is brittle in the face of refactorization (could be set with `nameof` which improves the situation)
* Requires either a `GetNode` call (brittle due to paths changing) or adding the node as a child, both of which *is* better but not ideal when this can be done another way more simply

We don't need anything more than straight linear interpolation so it can just track `MaxLife` and `RemainingLife` (lets set this initially to `MaxLife` in `SetShotPosition` which could be renamed to something like `Setup` and take a float arg for `MaxLife` so that after merging I can wire up higher damage shots to have longer lasting indicators). In `_Process` it can `RemainingLife -= Delta` (make sure to rename `delta` to `Delta`) and then feed `(RemainingLife / MaxLife) * 255)` in to the `Modulate`'s alpha channel.